### PR TITLE
 python3Packages.molecule{,-docker}: init

### PR DESCRIPTION
**Pull request by ilpianista**
###### Motivation for this change

Add `molecule`. Follow up of #138849.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by heph2** [comment]
I've spent the last 2 hours packaging molecule and molecule-docker and didn't check for opened PR...
BTW i've packaged molecule-hetznercloud and molecule-vagrant. I'm waiting that this is merged, then i'll make a PR too :D

**Comment by noisersup** [comment]
any updates about that?

**Comment by ilpianista** [comment]
I rebased it because ansible-compat has been added by #166283

**Review by dotlambda — CHANGES_REQUESTED** [review]
It would be nice to specify `format`, e.g. `format = "setuptools"`.

And please either add `pytestCheckHook` to `checkInputs`, specify `checkPhase`, or set `doCheck = false` with the reason as an inline comment.

**Comment by ilpianista** [comment]
> It would be nice to specify `format`, e.g. `format = "setuptools"`.
> 
> And please either add `pytestCheckHook` to `checkInputs`, specify `checkPhase`, or set `doCheck = false` with the reason as an inline comment.

Thank you! Please have a review whenever possible :pray:

**Comment by thornycrackers** [comment]
@dotlambda is there anything missing to get this merged in? Does it just need to be rebased?

**Comment by Et7f3** [comment]
>       > ERROR: Could not find a version that satisfies the requirement enrich>=1.2.5 (from molecule) (from versions: none)
>       > ERROR: No matching distribution found for enrich>=1.2.5

Can you update it ?

**Comment by dawidd6** [comment]
@ilpianista do you still have any interest left in advancing this PR further? I can have a stab at it.

File: pkgs/development/python-modules/molecule/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  pname = "molecule";
+  version = "3.5.2";
+  format = "setuptools";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1jp0djj69w0vw2697yr471zdkvd4smggcfvan5z8v6f0wncz0an8";
+  };
+
+  postPatch = ''
```
Comment (tboerger) [inline pkgs/development/python-modules/molecule/default.nix:29]: Maybe it is more useful to use [pythonRelaxDeps](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/python/hooks/python-relax-deps-hook.sh), as an example it's used at [ansible-doctor](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/admin/ansible/doctor.nix).

**Review by tboerger — COMMENTED** [review]
Should this really get into python-modules? These are not just modules, these are programs directly executable.

**Comment by ilpianista** [comment]
> @ilpianista do you still have any interest left in advancing this PR further? I can have a stab at it if anything.

molecule-docker is deprecated, and replaced by molecule-plugins (which provides all the drivers in different packages). I've tried to package them, but failed as I lack the required knowledge. Please take this over :pray:

**Comment by dawidd6** [comment]
@ilpianista opened a PR :slightly_smiling_face:

**Pull request closed without merging**